tests/queries/0_stateless/01079_new_range_reader_segfault.sql

```diff
@@ -3,7 +3,7 @@ drop table if exists t;
 create table t (a Int) engine = MergeTree order by a;
 
 -- some magic to satisfy conditions to run optimizations in MergeTreeRangeReader
-insert into t select number < 20 ? 0 : 1 from numbers(50);
+insert into t select number < 20000 ? 0 : 1 from numbers(50000);
 alter table t add column s String default 'foo';
 
 select s from t prewhere a != 1 where rand() % 2 = 0 limit 1;
```

Comment:
maybe modify condition in to smth determenistic like `where rowNumberInBlock() % 2 = 0`?

Reply:
Good idea, done.